### ISSUE TITLE
sys::Wait may use closed handle on Windows when process statistics are requested

### ISSUE THREAD

**Issue by andykaylor** [llvm/llvm-project#59979]
Support for process statistics was added to the sys::Wait function in this patch: https://reviews.llvm.org/D78901

If process statistics are needed, this code uses `GetProcessTimes(PI.Process, &CreationTime, &ExitTime, &KernelTime, &UserTime)` to get the program execution time. However, there are conditions under which this code may be reached after the `PI.Process` handle has been closed.

```
  DWORD WaitStatus = WaitForSingleObject(PI.Process, milliSecondsToWait);
  if (WaitStatus == WAIT_TIMEOUT) {
    if (SecondsToWait) {
      if (!TerminateProcess(PI.Process, 1)) {
        if (ErrMsg)
          MakeErrMsg(ErrMsg, "Failed to terminate timed-out program");

        // -2 indicates a crash or timeout as opposed to failure to execute.
        WaitResult.ReturnCode = -2;
        CloseHandle(PI.Process);
        return WaitResult;
      }
      **WaitForSingleObject(PI.Process, INFINITE);
      CloseHandle(PI.Process);**
    } else {
      // Non-blocking wait.
      return ProcessInfo();
    }
  }
```

I haven't seen this happen in a real scenario. It was reported by a static analysis tool.
